### PR TITLE
Update methods using the cylinder key-load feature

### PR DESCRIPTION
**Pull request by isabeltomb**
Update the `create_cylinder_jwt_auth` method to use the updated cylinder key load API to retrieve the private key used in jwt signing.

Update the default location that `keygen` saves keys to be ~/.cylinder/keys

Modify gameroom daemon to use the new `load_key` method from the cylinder key::load module.

Modify the scabbard `create_cylinder_jwt_auth` method to use the new methods in the cylinder key::load module.

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ path = "src/main.rs"
 
 [dependencies]
 clap = "2"
-cylinder = "0.2"
+cylinder = "^0.2.2"
```
Comment (vaporos) [inline cli/Cargo.toml:33]: Specifying the ^ here is unnecessary as that is implied in "0.2.2".

Comment (isabeltomb) [reply cli/Cargo.toml]: Fixed

**Review by ltseeley — APPROVED** [review]

File: cli/src/action/mod.rs [hunk 2]
```diff
@@ -134,26 +137,21 @@ fn msg_from_io_error(err: IoError) -> String {
...
         } else {
-            return Err(CliError::ActionError(
-                "Path is not valid unicode".to_string(),
-            ));
+            load_key(key_name, &current_user_search_path())
+                .map_err(|err| CliError::ActionError(format!("Unable to retrieve key: {}", err)))?
+                .ok_or_else(|| CliError::ActionError("Key not found".to_string()))?
         }
+    } else {
+        load_key(&current_user_key_name(), &current_user_search_path())
```
Comment (vaporos) [inline cli/src/action/mod.rs:150]: This needs to add ~/.splinter/keys to the search path to be consistent with keygen

**Review by vaporos — CHANGES_REQUESTED** [review]

File: cli/src/action/mod.rs (hunk 2 above)
Comment (isabeltomb) [reply cli/src/action/mod.rs]: Fixed

**Review by ltseeley — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**